core/runtime/src/main/java/io/quarkus/runtime/logging/LogMetricsHandler.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.runtime.logging;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+
+import org.jboss.logmanager.Level;
+
+public class LogMetricsHandler extends Handler {
+
+    AtomicLong defaultLogCounter;
+    Map<Level, AtomicLong> logCounters;
+
+    public LogMetricsHandler(Map<Level, AtomicLong> logCounters, AtomicLong defaultLogCounter) {
+        this.logCounters = logCounters;
+        this.defaultLogCounter = defaultLogCounter;
+    }
+
+    @Override
+    public void publish(LogRecord record) {
+        AtomicLong counter = logCounters.get(record.getLevel());
+        if (counter != null) {
+            counter.incrementAndGet();
+        } else {
+            defaultLogCounter.incrementAndGet();
```

Comment:
I just noticed the logic is wrong, should be renamed totalLogCounter and always incremented.
i.e. it should always be: vendor_log_total >= sum(vendor_log_total{level="*"})

Reply:
Do we really need a separate metric for the total when it is possible to compute that by adding together all other metrics?

Reply:
Indeed Grafana can take care of the sum, it's best no duplicate data.
But should we ignore events with non-standard levels (that's what Prometheus client is doing), or use a level="other" ?

Reply:
I've ignored the non-standard levels